compiler/src/locals.rs

```diff
@@ -1,20 +1,30 @@
-use analyzer::relations::LocalId;
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+
+use analyzer::relations::{LocalId, ResolvedSymbol};
+use analyzer::types::hir::Var;
 
 use crate::r#type::ValueStackSize;
 
 /// contains the different index per local value allocated in the locals area
 pub struct LocalsLayout {
     /// the start indexes of bound Locals
-    indexes: Vec<Option<u32>>,
+    values_indexes: Vec<Option<u32>>,
+    /// the start indexes of bound external values
+    external_refs_indexes: HashMap<ResolvedSymbol, u32>,
```

Comment:
The whole point of captures is to retrieve external symbols, that are firstly references, then copies. You can't keep track of that here, you need to follow along if the referred variable is dropped (because its scope ended) to transform the upvalue into a copy.